sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/models/WebPubSubAuthenticationToken.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.messaging.webpubsub.models;
+
+/**
+ * A wrapper class for results of the {@code getAuthenticationToken} APIs on
+ * {@link com.azure.messaging.webpubsub.WebPubSubAsyncServiceClient} and
+ * {@link com.azure.messaging.webpubsub.WebPubSubServiceClient}.
+ */
+public final class WebPubSubAuthenticationToken {
+    private final String authToken;
+    private final String url;
+
+    /**
+     * Creates a new instance with the given values set as immutable properties.
+     *
+     * @param authToken The authentication token that may be used to access the service.
+     * @param url The url that may be used to access the service.
+     */
+    public WebPubSubAuthenticationToken(String authToken, String url) {
```

Comment:
Is this functionality we expect to be common across SDKs?

Reply:
I haven't seen any other services needing this type of auth policy. @JonathanGiles may know if other client libraries would need it. If yes, we should consider moving to core.